hw2/task03/hw2_task03.py

```diff
@@ -0,0 +1,21 @@
+"""Write a function.
+
+That takes K lists as arguments and returns all possible
+lists of K items where the first element is from the first list,
+the second is from the second and so one.
+You may assume that that every list contain at least one element
+Example:
+assert comb([1, 2], [3, 4]) == [
+    [1, 3],
+    [1, 4],
+    [2, 3],
+    [2, 4],
+]
+"""
+
+
+def combination_of_lists(*args: list) -> list:
+    result = [[]]
+    for pool in args:
+        result = [base + [suffix] for base in result for suffix in pool]
```

Comment:
Optional question: how this approach is called in general?

Reply:
List comprehension and nested loops?

Reply:
I mean more general term, like `recursion`.